--- nas-1.6/clients/audio/auedit/Graph.c.64bit-fixes	1999-10-24 19:45:19.000000000 +0200
+++ nas-1.6/clients/audio/auedit/Graph.c	2003-08-04 12:18:59.000000000 +0200
@@ -161,7 +161,7 @@ GraphWidget     w;
 unsigned int    which;
 {
     XGCValues       gcv;
-#ifndef __alpha
+#if !defined(__alpha__) && !defined(__x86_64__) && !defined(__ia64__) && !defined(__LP64__)
     unsigned long   commonMask;
 #else /* __alpha */
     unsigned int  commonMask;
--- nas-1.6/lib/audio/Amd.h.64bit-fixes	1999-09-05 04:43:01.000000000 +0200
+++ nas-1.6/lib/audio/Amd.h	2003-08-04 12:19:18.000000000 +0200
@@ -120,7 +120,7 @@ pragma off(char_default_unsigned);
 #define B16
 #endif
 
-#ifndef __alpha
+#if !defined(__alpha__) && !defined(__x86_64__) && !defined(__ia64__) && !defined(__LP64__)
 typedef long		INT32;
 typedef unsigned long	CARD32;
 typedef unsigned long	BITS32;
